File: articles/meaning-and-purpose-positive-psychology/cpp/meaning_indices.cpp
#include <iostream>

struct MeaningIndicators {
    double meaning_presence;
    double meaning_search;
    double purpose_score;
    double coherence_score;
    double significance_score;
    double belonging_score;
    double value_alignment;
    double institutional_support;
    double wellbeing_score;
    double goal_persistence;
    double stress_load;
    double alienation_score;
    double identity_integration;
    double context_quality;
};

double meaning_system_index(const MeaningIndicators& x) {
    return (
        x.meaning_presence +
        x.purpose_score +
        x.coherence_score +
        x.significance_score +
        x.belonging_score +
        x.value_alignment +
        x.identity_integration
    ) / 7.0;
}

double context_adjusted_meaning(const MeaningIndicators& x) {
    return meaning_system_index(x) + x.institutional_support + x.context_quality - x.stress_load - x.alienation_score;
}

double directed_life_index(const MeaningIndicators& x) {
    return x.purpose_score + x.goal_persistence + x.value_alignment + x.institutional_support - x.stress_load;
}

double search_context_index(const MeaningIndicators& x) {
    return x.meaning_search + x.stress_load + x.alienation_score - x.meaning_presence - x.coherence_score;
}

int main() {
    MeaningIndicators example{7.1, 4.9, 7.2, 6.9, 7.1, 7.2, 7.0, 6.8, 7.1, 7.2, 3.7, 3.1, 6.9, 7.0};

    std::cout << "Meaning system index: " << meaning_system_index(example) << "\n";
    std::cout << "Context-adjusted meaning: " << context_adjusted_meaning(example) << "\n";
    std::cout << "Directed-life index: " << directed_life_index(example) << "\n";
    std::cout << "Search-context index: " << search_context_index(example) << "\n";

    return 0;
}

File: articles/meaning-and-purpose-positive-psychology/fortran/meaning_indices.f90
program meaning_indices
  implicit none
  real :: meaning_presence, meaning_search, purpose_score, coherence_score
  real :: significance_score, belonging_score, value_alignment, institutional_support
  real :: wellbeing_score, goal_persistence, stress_load, alienation_score
  real :: identity_integration, context_quality
  real :: meaning_system, context_adjusted, directed_life, search_context

  meaning_presence = 7.1
  meaning_search = 4.9
  purpose_score = 7.2
  coherence_score = 6.9
  significance_score = 7.1
  belonging_score = 7.2
  value_alignment = 7.0
  institutional_support = 6.8
  wellbeing_score = 7.1
  goal_persistence = 7.2
  stress_load = 3.7
  alienation_score = 3.1
  identity_integration = 6.9
  context_quality = 7.0

  meaning_system = (meaning_presence + purpose_score + coherence_score + significance_score + &
       belonging_score + value_alignment + identity_integration) / 7.0
  context_adjusted = meaning_system + institutional_support + context_quality - stress_load - alienation_score
  directed_life = purpose_score + goal_persistence + value_alignment + institutional_support - stress_load
  search_context = meaning_search + stress_load + alienation_score - meaning_presence - coherence_score

  print *, "Meaning system index:", meaning_system
  print *, "Context-adjusted meaning:", context_adjusted
  print *, "Directed-life index:", directed_life
  print *, "Search-context index:", search_context
end program meaning_indices
